### PR TITLE
Us26+27

### LINKED ISSUE #26: Merchant Invoice Show Page
As a merchant
When I visit my merchant's invoice show page(/merchants/merchant_id/invoices/invoice_id)
Then I see information related to that invoice including:
- Invoice id
- Invoice status
- Invoice created_at date in the format "Monday, July 18, 2019"
- Customer first and last name

### LINKED ISSUE #27: Merchant Invoices Index
As a merchant,
When I visit my merchant's invoices index (/merchants/merchant_id/invoices)
Then I see all of the invoices that include at least one of my merchant's items
And for each invoice I see its id
And each id links to the merchant invoice show page

### PR DESCRIPTION
**Pull request by jgrazulis**
## Description of Changes

- Added model methods for creation_date_formatted
- Added model tests for said method
- Added test data missing for item and invoice item (description, unit price)
- Added show and index views 
- Added show method to controller 

## Concerns (tags recommended)

- Looks like there will be some merge conflicts since I started working on these this morning before merges
- Date tests are not dynamic since they go with the current date, which for our test data is today

### CODE REVIEW

File: app/models/invoice.rb [hunk 1]
```diff
@@ -11,6 +11,15 @@ def self.incomplete_invoices
     joins(:invoice_items).where('invoice_items.status != ?', 2).distinct
   end
 
+  def creation_date_formatted
+    date = self.created_at
+    date.strftime("%A, %B %d, %Y")
+  end
+
+  def items_by_merchant(merchant_id)
```
Comment (glsorensen) [inline app/models/invoice.rb:19]: Nice AR

File: app/views/merchant_invoices/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<h1>Invoice Number <%= @invoice.id %></h1>
+
+<h2>Invoice Status: <%= @invoice.status %></h2>
+<h2>Created On: <%= @invoice.creation_date_formatted %></h2>
+<h2>Customer Information:</h2>
+<p>First Name: <%= @invoice.customer.first_name %></p>
+<p>Last Name: <%= @invoice.customer.last_name %></p>
```
Comment (glsorensen) [inline app/views/merchant_invoices/show.html.erb:7]: This is a meaty one, nice use if divs

**Review by glsorensen — APPROVED** [review]
No issues with the merge conlfict, I took care of it. Looks good

File: app/views/merchant_invoices/show.html.erb (hunk 2 above)
Comment (jgrazulis) [reply app/views/merchant_invoices/show.html.erb:7]: Thanks!

File: app/models/invoice.rb [hunk 3]
```diff
@@ -11,6 +11,15 @@ def self.incomplete_invoices
     joins(:invoice_items).where('invoice_items.status != ?', 2).distinct
   end
 
+  def creation_date_formatted
```
Comment (gregoryjflaherty) [inline app/models/invoice.rb:14]: Yeah I really like this being in a method - we can go back later and reuse this in a bunch of places I believe

File: spec/models/invoice_spec.rb [hunk 4]
```diff
@@ -33,12 +35,14 @@
       @invoice_5 = Invoice.create(customer_id: @customer_2.id, status: 1)
       @invoice_6 = Invoice.create(customer_id: @customer_2.id, status: 1, created_at: '1989-04-02')
 
-      @invoice_item_1 = InvoiceItem.create(invoice_id: @invoice_1.id, item_id: @shoe_1.id, status: 2)
-      @invoice_item_2 = InvoiceItem.create(invoice_id: @invoice_1.id, item_id: @shoe_2.id, status: 2)
-      @invoice_item_3 = InvoiceItem.create(invoice_id: @invoice_1.id, item_id: @shoe_3.id, status: 2)
-      @invoice_item_4 = InvoiceItem.create(invoice_id: @invoice_2.id, item_id: @shoe_4.id, status: 1)
-      @invoice_item_5 = InvoiceItem.create(invoice_id: @invoice_2.id, item_id: @shoe_5.id, status: 1)
-      @invoice_item_6 = InvoiceItem.create(invoice_id: @invoice_2.id, item_id: @shoe_6.id, status: 1)
```
Comment (gregoryjflaherty) [inline spec/models/invoice_spec.rb:41]: These tests can be a pain to deal with - good job testing thoroughly

**Review by gregoryjflaherty — APPROVED** [review]

**Pull request merged**